gossipsub_test.go

```diff
@@ -297,14 +297,14 @@ func TestGossipsubFanoutExpiry(t *testing.T) {
 		}
 	}
 
-	if len(psubs[0].rt.(*GossipSubRouter).fanout) == 0 {
+	if len(psubs[0].rt.(*GossipConfigurableRouter).fanout) == 0 {
```

Comment:
Since NewGossipSub return a PubSub instance these type assertions were the only places where I ran into issues with renaming GossipSubRouter into GossipConfigurableRouter. However, if we think there are more places where this could be an issue we could utilize a modified GossipSubRouter that is based on a GossipConfigurableRouter.

Reply:
Please don't change the name of the router. It's the gossipsub router, and it has become configurable.

Reply:
no problem